src/protocols/bdx/tests/TestTransferDiagnosticLog.cpp

```diff
@@ -0,0 +1,366 @@
+    transferInitData.StartOffset      = 0;
+    transferInitData.Length           = 1024;
+    char testFileDes[9]               = { "test.txt" };
+    transferInitData.FileDesLength    = 9;
+    transferInitData.FileDesignator   = reinterpret_cast<uint8_t *>(testFileDes);
+    uint8_t fakeData[5]               = { 7, 6, 5, 4, 3 };
+    transferInitData.MetadataLength   = 5;
+    transferInitData.Metadata         = reinterpret_cast<uint8_t *>(fakeData);
+
+    /// Init initiator (and sender) transfer session
+    auto r = initiator.StartTransfer(TransferRole::kSender, transferInitData, System::Clock::Seconds16(10));
+
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    TransferSession::OutputEvent initiatorEvent;
+    initiator.PollOutput(initiatorEvent, System::Clock::kZero);
+    EXPECT_EQ(initiatorEvent.EventType, TransferSession::OutputEventType::kMsgToSend);
+
+    /// Init responder (and receiver) transfer session
+    r = mTransferSession.WaitForTransfer(TransferRole::kReceiver, TransferControlFlags::kSenderDrive, kMaxBlockSize,
+                                         System::Clock::Seconds16(20));
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    TransferSession::OutputEvent responderEvent;
+    mTransferSession.PollOutput(responderEvent, System::Clock::kZero);
+    EXPECT_EQ(responderEvent.EventType, TransferSession::OutputEventType::kNone);
+
+    chip::PayloadHeader payloadHeader;
+    payloadHeader.SetMessageType(initiatorEvent.msgTypeData.ProtocolId, initiatorEvent.msgTypeData.MessageType);
+
+    r = mTransferSession.HandleMessageReceived(payloadHeader, std::move(initiatorEvent.MsgData), System::Clock::kZero);
+
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    mTransferSession.PollOutput(responderEvent, System::Clock::kZero);
+
+    EXPECT_EQ(responderEvent.EventType, TransferSession::OutputEventType::kInitReceived);
+
+    r = proxyDiagnosticLog.Init(&mTransferSession);
+
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    proxyDiagnosticLog.SetFabricIndex(kFabricIndex);
+    proxyDiagnosticLog.SetPeerNodeId(kNodeId);
+    r = proxyDiagnosticLog.Accept();
+
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    mTransferSession.PollOutput(responderEvent, System::Clock::kZero);
+
+    EXPECT_EQ(responderEvent.EventType, TransferSession::OutputEventType::kMsgToSend); // responder sends Accept message
+
+    /// Now, initiator handles SendAccept message
+    payloadHeader.SetMessageType(responderEvent.msgTypeData.ProtocolId, responderEvent.msgTypeData.MessageType);
+
+    r = initiator.HandleMessageReceived(payloadHeader, std::move(responderEvent.MsgData), System::Clock::kZero);
+
+    EXPECT_EQ(r, CHIP_NO_ERROR);
+
+    initiator.PollOutput(initiatorEvent, System::Clock::kZero);
+
+    EXPECT_EQ(initiatorEvent.EventType, TransferSession::OutputEventType::kAcceptReceived);
+
+    /// initiator prepares block to send
+    TransferSession::BlockData blockData;
+    blockData.Data         = reinterpret_cast<const uint8_t *>("hi there");
```

Comment:
do we care about the content? I would rather try to reduce casts, so maybe some {1,2,3,4,5} or something for a constant would be less casting.